src/app/public/components/public-archive/public-archive.component.spec.ts

```diff
@@ -54,4 +56,34 @@ describe('PublicArchiveComponent', () => {
       'archive-description-show': true,
     });
   });
+
+  it('should navigate to the correct search URL on handleSearch', async () => {
+    const { instance, inject } = await shallow.render();
+    const router = inject(Router);
+    spyOn(router, 'navigate');
+
+    instance.archive = { archiveId: '123' } as any;
+
+    instance.onHandleSearch('test-query');
+
+    expect(router.navigate).toHaveBeenCalledWith(
+      ['search', '123', 'test-query'],
+      { relativeTo: instance.route },
```

Comment:
I'm not sure we should be exposing the `ActivatedRoute` of this component publicly just for the sake of unit testing convenience. Can you try constructing the `router.navigate` test double in such a way where we can ignore that other parameter? I don't think we need to be so granular as to test the existence of the `relativeTo` property.